lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
```

Comment:
'service' is always nil in this loop. I.e., it never matches. Moreover, since we are in a branch where driver_services.size == 1 AND service == nil, the matched service is indeed guaranteed to be driver_services.first